halo2_proofs/src/plonk.rs

```diff
@@ -120,9 +120,21 @@ where
                 "unexpected version byte",
             ));
         }
-        let mut k = [0u8; 4];
+        // Maximum allowed value for parameter `k`, the log-size of the circuit.
+        const MAX_CIRCUIT_SIZE: u8 = 32;
+
+        let mut k = [0u8; 1];
```

Comment:
Are you sure we are serializing it as u8 also? JIC we are serializing with 4 bytes in other places :worried: